api/access/server.go

```diff
@@ -128,12 +129,87 @@ func (s *Server) GetLatestBlock(ctx context.Context, in *access.GetLatestBlockRe
 	return nil, errors.New("not implemented")
 }
 
-func (s *Server) GetBlockByID(ctx context.Context, in *access.GetBlockByIDRequest) (*access.BlockResponse, error) {
-	return nil, errors.New("not implemented")
+func (s *Server) GetBlockByID(_ context.Context, in *access.GetBlockByIDRequest) (*access.BlockResponse, error) {
+	blockID := flow.HashToID(in.Id)
+	height, err := s.index.HeightForBlock(blockID)
+	if err != nil {
+		return nil, fmt.Errorf("could not get height for block ID %x: %w", blockID, err)
+	}
+
+	header, err := s.index.Header(height)
+	if err != nil {
+		return nil, fmt.Errorf("could not get header for height %d: %w", height, err)
+	}
+
+	ss, err := s.index.SealsByHeight(height)
```

Comment:
It's funny that you are now starting to make the naming so short as to be confusing :D

`sealIDs` and `sealID` as the loop variable seem as the natural choice here.